access/grpc/grpc.go

```diff
@@ -1188,14 +1188,10 @@ func (c *BaseClient) SubscribeBlocksFromStartBlockID(
 		return nil, nil, newRPCError(err)
 	}
 
-	blocksChan := make(chan flow.Block)
-	errChan := make(chan error)
-
-	go func() {
-		defer close(blocksChan)
-		defer close(errChan)
-		receiveBlocksFromClient(ctx, subscribeClient, blocksChan, errChan)
-	}()
+	convertBlockResponse := func(response *access.SubscribeBlocksResponse) (flow.Block, error) {
+		return convert.MessageToBlock(response.GetBlock())
+	}
+	blocksChan, errChan := subscribe(ctx, subscribeClient.Recv, convertBlockResponse, "block")
```

Comment:
you could reduce a little more boilerplate by having `subscribe` return a nil error and just returning the result directly
```suggestion
	return subscribe(ctx, subscribeClient.Recv, convertBlockResponse, "block")
```

Reply:
Actually, we return 3 values from the function `(chan, chan, error)`, so I cannot change it. 
I cannot even write `subscribe(), nil` due to language syntax restrictions